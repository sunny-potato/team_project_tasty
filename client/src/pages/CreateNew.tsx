import React, { useState } from 'react';
import { RecipeInfo, Ingredient } from '../DataService';
import InputRecipeInfo from '../components/InputRecipeInfo';
import InputIngredients from '../components/InputIngredients';
import { useNavigate } from 'react-router-dom';
import { ValidateData, SaveEditedData } from '../components/SaveInputData';

const initialRecipeInfo: RecipeInfo = {
  id: undefined!,
  name: '',
  meal_type: '',
  new: false,
  popular: false,
  description: '',
};

const initialIngredient: Ingredient = {
  ingredients_id: undefined!,
  ingredient: '',
  amount: null,
  unit_id: 0,
  unit: 'initalUnit',
};

export function CreateNew() {
  const navigate = useNavigate();
  const [recipeInfo, setRecipeInfo] = useState<RecipeInfo>(initialRecipeInfo);
  const [ingredients, setIngredients] = useState<Ingredient[]>([initialIngredient]);
  const [isLoading, setIsLoading] = useState<boolean>(false);

  // console.log(recipeInfo);
  // console.log(ingredients);
  const onSubmit = (event: React.FormEvent) => {
    event.preventDefault();

    if (!isLoading) {
      setIsLoading(true);
      const isValid = ValidateData({ ingredients }); // current condition is always true
      if (isValid) {
        SaveEditedData({ ingredients, recipeInfo });
      }
    } else {
      alert('something wrong');
      //*********************//
      //here will be added f meal_type or description is required
    }
    setIsLoading(false);
  };

  return (
    <div>
      <h1 className="Page-title">Create new recipe</h1>
      <div className="Content-main">
        <form onSubmit={onSubmit}>
          <InputRecipeInfo recipeInfo={recipeInfo} setRecipeInfo={setRecipeInfo} />
          <InputIngredients ingredients={ingredients} setIngredients={setIngredients} />
          <div className="btn-group">
            <button type="submit" style={{ opacity: isLoading ? 0.3 : 1 }}>
              {isLoading ? 'Saving' : 'Save'}
            </button>
            <button type="button" className="cancelButton" onClick={() => navigate('/')}>
              Cancel
            </button>
          </div>
        </form>
      </div>
    </div>
  );
}
