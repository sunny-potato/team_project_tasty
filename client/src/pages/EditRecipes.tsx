import React from 'react';
import { useParams } from 'react-router-dom';

export function EditRecipes() {
  return (
    <div>
      <h1>Site to edit the recipes</h1>
    </div>
  );
}
