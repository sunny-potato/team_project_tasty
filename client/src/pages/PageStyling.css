@import url('https://fonts.googleapis.com/css2?family=Prompt:wght@200;600&display=swap');

* {
  box-sizing: border-box;
  margin: 0;
  padding: 0;
}

body {
  background-color: white;
  padding: 0;
  text-align: center;
}
.Content-main {
  display: flex;
  flex-direction: column;
  align-items: center;
  margin: 1rem;
  padding: 1rem;
}
p {
  max-width: 750px;
  margin: 1.5rem;
  padding: 1rem;
}
.Ingredients-main {
  width: max-content;
  margin-left: auto;
  margin-right: auto;
  margin-top: 1.5rem;
}
.Ingredients-header {
  padding: 1rem 1rem;
  text-align: left;
}
.Ingredients-cell {
  padding: 0.2rem;
  text-align: left;
}
.Ingredients-name {
  text-align: left;
}
.Recipe-popular {
  margin: 0.2rem;
  background-color: rgba(255, 182, 139, 0.377);
  font-size: 0.8rem;
  padding: 0.3rem;
  border-radius: 2rem;
  font-family: 'Prompt';
  font-weight: 200;
}
.Recipe-new {
  content: none;
  margin: 0.4rem;
  background-color: rgba(125, 230, 139, 0.452);
  font-size: 0.8rem;
  padding: 0.3rem;
  border-radius: 2rem;
  font-family: 'Prompt';
  font-weight: 200;
}
.Recipe-tags {
  margin: 0.5rem;
}
.Button-navigation {
  padding: 0.5rem;
  margin: 1.5rem;
  width: fit-content;
  background-color: rgb(41, 211, 118);
  border: none;
  border-radius: 3px;
  font-family: 'Prompt';
  font-weight: 500;
  cursor: pointer;
}
.Button-navigation:hover {
  background-color: rgba(41, 211, 117, 0.418);
}

.Button-action {
  padding: 0.5rem;
  margin: 2.5rem;
  background-color: rgb(41, 211, 118);
  border: none;
  border-radius: 3px;
  font-family: 'Prompt';
  font-weight: 500;
  cursor: pointer;
}
.Button-action:hover {
  background-color: rgba(41, 211, 117, 0.418);
}
.User-input {
  padding: 0;
  margin: 1rem;
}
.Button-recipe-navigation {
  padding: 1rem;
  margin: 1.5rem;
  width: 20rem;
  background-color: rgb(41, 211, 118);
  border: none;
  border-radius: 3px;
  font-family: 'Prompt';
  font-weight: 500;
  cursor: pointer;
}
.Button-recipe-navigation:hover {
  background-color: rgba(41, 211, 117, 0.418);
}
