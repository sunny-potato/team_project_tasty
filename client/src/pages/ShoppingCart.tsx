import React from 'react';

export function ShoppingCart() {
  return (
    <div>
      <h1>This is your shopping cart to be....</h1>
    </div>
  );
}
