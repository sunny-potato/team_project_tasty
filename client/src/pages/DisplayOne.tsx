import './PageStyling.css';
import React, { useEffect, useState, useRef, Component } from 'react';
import dataService, { Recipe, Ingredient, RecipeInfo } from '../DataService';
import { Link, useParams, useLocation } from 'react-router-dom';

export function DisplayOne() {
  const [recipe, setRecipe] = useState<Recipe>();
  const [ingredients, setIngredients] = useState<Ingredient[]>();

  //To get navigation by page and id
  const params = useParams();
  const id: string = params.id;

  // Get inital value in load, uses recipe id as source
  // Also loads values recipe and ingredients by setState

  useEffect(() => {
    let id: string = params.id;

    dataService.get(id).then((data) => {
      setRecipe(data);
      setIngredients(data.ingredients);
    });
  }, []);

  // Updates ingredients according to value chosen by user

  function changePortions(portions: number) {
    let newAmounts: Ingredient[];
    let defaultAmounts: Ingredient[] = recipe!.ingredients;

    newAmounts = defaultAmounts?.map((i) => ({
      ...i,
      amount: parseFloat(((i.amount / 4) * portions).toPrecision(2)),
    }));
    setIngredients(newAmounts);
  }

  return (
    <div className="Content-main">
      <h1 className="Recipe-title">{recipe?.recipeInfo.name}</h1>
      <h5 className="Recipe-tags">Tags: {recipe?.recipeInfo.meal_type}</h5>
      <h6 className="Recipe-new">{recipe?.recipeInfo.new ? 'This is a new recipe!' : null}</h6>
      <h6 className="Recipe-popular">
        {recipe?.recipeInfo.popular ? 'This item is popular right now!' : null}
      </h6>
      <p className="Recipe-description>">{recipe?.recipeInfo.description}</p>
      <div>
        <div>
          <label>Number of portions:&nbsp;</label>
          <select
            name="portions"
            defaultValue={4}
            onChange={(e) => changePortions(parseInt(e.currentTarget.value))}
          >
            <option value={1}>1</option>
            <option value={2}>2</option>
            <option value={3}>3</option>
            <option value={4}>4</option>
            <option value={5}>5</option>
            <option value={6}>6</option>
            <option value={7}>7</option>
            <option value={8}>8</option>
            <option value={9}>9</option>
            <option value={10}>10</option>
          </select>
        </div>
        <table className="Ingredients-main">
          <tbody>
            <tr>
              <th className="Ingredients-header">Ingredients:</th>
              <th></th>
            </tr>
            {ingredients?.map((content) => (
              <tr className="Ingredients-row" key={content.ingredients_id}>
                <td className="Ingredients-cell">
                  {content.amount}&nbsp;{content.unit}
                </td>

                <td className="Ingredients-name">{content.ingredient}</td>
              </tr>
            ))}
          </tbody>
        </table>
        <Link to="/">
          <button className="Button-navigation">Back</button>
        </Link>
        <Link to="/edit">
          <button className="Button-navigation">Edit recipe</button>
        </Link>
        <button
          className="Button-navigation"
          onClick={() => alert('Here will the like function be, not implemented yet')}
        >
          Like
        </button>
      </div>
    </div>
  );
}
