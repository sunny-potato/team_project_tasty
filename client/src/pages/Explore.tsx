import "./PageStyling.css";
import React, { useEffect, useState} from "react";
import dataService, { Recipe, Ingredient, RecipeInfo } from "../DataService";
import { NavLink } from "react-router-dom";

export function Explore() {

  const [items, setItems] = useState<Recipe[]>([]);


    // Get inital value in load, uses recipe id as source
  useEffect(() => {
    getItems();
  },[])

  //geting the API request, unless you already have one inside localStorage
  const getItems = () => {
    const check = localStorage.getItem('Items')
    
    if (check) {
      dataService.apiExploreData(JSON.parse(check)).then((data) => {
      setItems(data)
      });
    } else {

      dataService.apiExploreKey().then(async (key: string) => {
      const api:Response = await fetch(`https://api.spoonacular.com/recipes/random?apiKey=${key}&number=9`
      );
      const data = await api.json();
      localStorage.setItem('Items', JSON.stringify(data.recipes))
      
      dataService.apiExploreData(data.recipes).then((data) => {
        setItems(data)
      });

      });
    }  
  }

  //what the page sees
  return (
    <div className="Content-main">
      <h1>A place to discover new recipes, API or other</h1>
        {items?.map((recipe) => { 
          let recipeinfo:RecipeInfo = recipe.recipeInfo
          let ingrediens:Ingredient[] = recipe.ingredients
          
          return(
          <div key={recipeinfo.id}>
            <NavLink to={'/recipe/' + recipeinfo.id}>{recipeinfo.name}</NavLink>
            {ingrediens?.map((ingrediens) => { 
            })}
          </div>
          );

        })}
    </div>
  );
}
// <img src={recipe.image} alt={recipeinfo.name}></img>
