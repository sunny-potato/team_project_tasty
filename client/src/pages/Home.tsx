import React, { useState, useEffect } from 'react';
import { Link } from 'react-router-dom';
import dataService from '../DataService';
import '../css/PageStyling.css';
import { RecipeInfo, Recipe } from '../DataService';
import { ImCross } from 'react-icons/im';
import InputIngredients from 'src/components/InputIngredients';

export function Home() {
  //Get all recipes and update variable allRecipes with data
  const [allRecipes, setAllRecipes] = useState<Recipe[]>();
  const [queryList, setQueryList] = useState<Recipe[]>();
  const [filterList, setFilterList] = useState<Recipe[]>();
  const [query, setQuery] = useState('');
  const [filter, setFilter] = useState('');
  const [isClicked, setIsClicked] = useState(false);
  const tagKeys = ['Dinner', 'Lunch', 'Breakfast', 'Snack', 'Dessert', 'Popular', 'New'];
  useEffect(() => {
    dataService.getAll().then((data) => {
      setAllRecipes(data);
      setFilterList(data);
      setQueryList(data);
    });
  }, []);

  useEffect(() => {
    if (query.length === 0 || null || undefined) {
      // setQueryList(allRecipes);
    } else {
      // if (query.includes('popular')) {
      //   let search: Recipe[] | any = filterList?.filter((m) => m.recipeInfo.popular === true);
      //   setFilterList(search);
      // } else if (query.includes('new')) {
      //   let search: Recipe[] | any = filterList?.filter((m) => m.recipeInfo.new === true);
      //   setFilterList(search);
      // } else {
      let search: Recipe[] | any = filterList?.filter(
        (m) =>
          m.recipeInfo.name.toLowerCase().includes(query) ||
          m.recipeInfo.description.toLowerCase().includes(query) ||
          m.ingredients.some((i) => i.ingredient.includes(query))
      );

      setQueryList(search);
    }
    // }
  }, [query]);

  useEffect(() => {
    if (filter === '' || filter === undefined || filter === null) {
      setFilterList(allRecipes);
      setQueryList(allRecipes);
    } else if (filter.includes('popular')) {
      let filtered: Recipe[] | any = allRecipes?.filter((m) => m.recipeInfo.popular === true);
      setFilterList(filtered);
      setQueryList(filtered);
    } else if (filter.includes('new')) {
      let filtered: Recipe[] | any = allRecipes?.filter((m) => m.recipeInfo.new === true);
      setFilterList(filtered);
      setQueryList(filtered);
    } else {
      let filtered: Recipe[] | any = allRecipes?.filter((t) =>
        t.recipeInfo.meal_type.toLowerCase().includes(filter)
      );
      setFilterList(filtered);
      setQueryList(filtered);
    }
  }, [isClicked]);

  return (
    <div className="Content-main">
      <div>
        <h1>Welcome to Tasty!</h1>
        <h4>The home of recipes...</h4>
        <p>
          Here you can explore your own recipes by adding new content, editing existing ones and
          deleting the ones you don't care about anymore.
        </p>
        <p>Use the search and filtering function to narrow your search.</p>
      </div>
      <input
        className="Input-search"
        type="text"
        placeholder="Search recipes for name, ingredients and description...."
        value={query}
        onChange={(e) => {
          setQuery(e.currentTarget.value.toLowerCase());
        }}
      ></input>
      <h5 className="Extra-information">Filters:</h5>
      <div>
        {isClicked ? (
          <button
            className="Button-filter-active"
            key={filter}
            value={''}
            onClick={(e) => {
              setFilter('');
              setIsClicked(!isClicked);
            }}
          >
            <ImCross />
            &nbsp;{filter}
          </button>
        ) : (
          tagKeys.map((e) => (
            <button
              className="Button-filter"
              key={e}
              value={e}
              onClick={(e) => {
                setQuery('');
                setFilter(e.currentTarget.value.toLowerCase());
                setIsClicked(!isClicked);
              }}
            >
              {e}
            </button>
          ))
        )}
      </div>
      <div className="Content-second">
        {' '}
        {queryList ? (
          queryList.slice(0, 8).map((recipe: Recipe) => (
            <Link key={recipe.recipeInfo.id} to={'/recipe/' + recipe.recipeInfo.id}>
              <button className="Button-recipe-navigation">{recipe.recipeInfo.name}</button>
            </Link>
          ))
        ) : (
          <b>Oops...You have no recipes to show here</b>
        )}
      </div>
    </div>
  );
}
