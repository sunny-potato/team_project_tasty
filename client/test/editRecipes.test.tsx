import * as React from 'react';
import { shallow } from 'enzyme';
import { EditRecipes } from '../src/pages/EditRecipes';

describe('EditRecipes page tests', () => {
  test('EditRecipes page draws correctly', (done) => {
    const wrapper = shallow(<EditRecipes />);
    setTimeout(() => {
      expect(wrapper).toMatchSnapshot();
      done();
    });
  });
});
