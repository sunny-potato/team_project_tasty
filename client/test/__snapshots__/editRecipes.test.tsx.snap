// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditRecipes page tests Edit Recipe, page draws correctly (snapshot) 1`] = `
<div>
  <div>
    <h1
      class="Page-title"
    >
      Edit recipe
    </h1>
    <div
      class="Content-main"
    >
      <form>
        <div
          class="Content-input-container"
        >
          <div
            class="row"
          >
            <input
              autocomplete="off"
              name="name"
              placeholder="Name of the recipe"
              required=""
              type="text"
              value="One-Pot Mac and Cheese"
            />
          </div>
          <div
            class="row"
          >
            <p
              class="Tag"
            >
              Choose a tag
            </p>
            <p
              class="Tag"
            >
              <button
                class="Button-select-option"
                style="background-color: rgb(41, 211, 118); color: white; border: 1px solid black;"
                type="button"
              >
                Dinner
              </button>
              <button
                class="Button-select-option"
                style="background-color: white; color: black; border: 1px solid lightgrey;"
                type="button"
              >
                Lunch
              </button>
              <button
                class="Button-select-option"
                style="background-color: white; color: black; border: 1px solid lightgrey;"
                type="button"
              >
                Breakfast
              </button>
              <button
                class="Button-select-option"
                style="background-color: white; color: black; border: 1px solid lightgrey;"
                type="button"
              >
                Snack
              </button>
              <button
                class="Button-select-option"
                style="background-color: white; color: black; border: 1px solid lightgrey;"
                type="button"
              >
                Dessert
              </button>
            </p>
          </div>
          <div
            class="row"
          >
            <textarea
              autocomplete="off"
              cols="50"
              name="description"
              placeholder="Describe your recipe and how to make it delicious"
              rows="4"
            >
              Who likes cleaning up after making mac and cheese? 
              Not this girl. This one-pot mac and cheese is a family favorite, 
              and my 3-year-old is thrilled to see it coming to the dinner table. 
              We love to add sliced smoked sausage to this creamy mac recipe!
            </textarea>
          </div>
        </div>
        <div>
          <div
            class="row"
          >
            <label
              class="content-portions"
            >
              Number of portions :
               
              <select
                name="portions"
              >
                <option
                  value="1"
                >
                  1
                </option>
                <option
                  value="2"
                >
                  2
                </option>
                <option
                  value="3"
                >
                  3
                </option>
                <option
                  value="4"
                >
                  4
                </option>
                <option
                  value="5"
                >
                  5
                </option>
                <option
                  value="6"
                >
                  6
                </option>
                <option
                  value="7"
                >
                  7
                </option>
                <option
                  value="8"
                >
                  8
                </option>
                <option
                  value="9"
                >
                  9
                </option>
                <option
                  value="10"
                >
                  10
                </option>
              </select>
            </label>
          </div>
          <div
            class="row"
          >
            <table
              class="table-ingredients"
            >
              <tbody>
                <tr
                  class="table-headerRow"
                >
                  <th
                    colspan="3"
                  >
                    Ingredients
                  </th>
                  <th>
                    <button
                      class="Button-choice-add"
                      type="button"
                    >
                      + Add
                    </button>
                  </th>
                </tr>
                <tr
                  class="table-headerRow"
                >
                  <th>
                    Amount
                  </th>
                  <th>
                    Unit
                  </th>
                  <th>
                    Name
                  </th>
                  <th />
                </tr>
                <tr
                  class="table-contentRow"
                >
                  <td
                    class="Table-amount"
                  >
                    <div>
                      <input
                        autocomplete="off"
                        min="0"
                        name="amount"
                        step="any"
                        type="number"
                        value="1.5"
                      />
                    </div>
                  </td>
                  <td>
                    <div>
                      <select
                        autocomplete="off"
                        name="unit"
                      >
                        <option
                          value=""
                        />
                        <option
                          value="cup"
                        >
                          cup
                        </option>
                        <option
                          value="package"
                        >
                          package
                        </option>
                        <option
                          value="ounce"
                        >
                          ounce
                        </option>
                      </select>
                    </div>
                  </td>
                  <td>
                    <div>
                      <input
                        autocomplete="off"
                        name="ingredient"
                        placeholder="Ingredient"
                        required=""
                        type="text"
                        value="whole milk"
                      />
                    </div>
                  </td>
                  <td>
                    <button
                      class="Button-choice-delete"
                      type="button"
                    >
                      Delete
                    </button>
                  </td>
                </tr>
                <tr
                  class="table-contentRow"
                >
                  <td
                    class="Table-amount"
                  >
                    <div>
                      <input
                        autocomplete="off"
                        min="0"
                        name="amount"
                        step="any"
                        type="number"
                        value="3"
                      />
                    </div>
                  </td>
                  <td>
                    <div>
                      <select
                        autocomplete="off"
                        name="unit"
                      >
                        <option
                          value=""
                        />
                        <option
                          value="cup"
                        >
                          cup
                        </option>
                        <option
                          value="package"
                        >
                          package
                        </option>
                        <option
                          value="ounce"
                        >
                          ounce
                        </option>
                      </select>
                    </div>
                  </td>
                  <td>
                    <div>
                      <input
                        autocomplete="off"
                        name="ingredient"
                        placeholder="Ingredient"
                        required=""
                        type="text"
                        value="water"
                      />
                    </div>
                  </td>
                  <td>
                    <button
                      class="Button-choice-delete"
                      type="button"
                    >
                      Delete
                    </button>
                  </td>
                </tr>
                <tr
                  class="table-contentRow"
                >
                  <td
                    class="Table-amount"
                  >
                    <div>
                      <input
                        autocomplete="off"
                        min="0"
                        name="amount"
                        step="any"
                        type="number"
                        value="1"
                      />
                    </div>
                  </td>
                  <td>
                    <div>
                      <select
                        autocomplete="off"
                        name="unit"
                      >
                        <option
                          value=""
                        />
                        <option
                          value="cup"
                        >
                          cup
                        </option>
                        <option
                          value="package"
                        >
                          package
                        </option>
                        <option
                          value="ounce"
                        >
                          ounce
                        </option>
                      </select>
                    </div>
                  </td>
                  <td>
                    <div>
                      <input
                        autocomplete="off"
                        name="ingredient"
                        placeholder="Ingredient"
                        required=""
                        type="text"
                        value="elbow macaroni (16 ounces)"
                      />
                    </div>
                  </td>
                  <td>
                    <button
                      class="Button-choice-delete"
                      type="button"
                    >
                      Delete
                    </button>
                  </td>
                </tr>
                <tr
                  class="table-contentRow"
                >
                  <td
                    class="Table-amount"
                  >
                    <div>
                      <input
                        autocomplete="off"
                        min="0"
                        name="amount"
                        step="any"
                        type="number"
                        value="4"
                      />
                    </div>
                  </td>
                  <td>
                    <div>
                      <select
                        autocomplete="off"
                        name="unit"
                      >
                        <option
                          value=""
                        />
                        <option
                          value="cup"
                        >
                          cup
                        </option>
                        <option
                          value="package"
                        >
                          package
                        </option>
                        <option
                          value="ounce"
                        >
                          ounce
                        </option>
                      </select>
                    </div>
                  </td>
                  <td>
                    <div>
                      <input
                        autocomplete="off"
                        name="ingredient"
                        placeholder="Ingredient"
                        required=""
                        type="text"
                        value="velveeta, cubed"
                      />
                    </div>
                  </td>
                  <td>
                    <button
                      class="Button-choice-delete"
                      type="button"
                    >
                      Delete
                    </button>
                  </td>
                </tr>
                <tr
                  class="table-contentRow"
                >
                  <td
                    class="Table-amount"
                  >
                    <div>
                      <input
                        autocomplete="off"
                        min="0"
                        name="amount"
                        step="any"
                        type="number"
                        value="2"
                      />
                    </div>
                  </td>
                  <td>
                    <div>
                      <select
                        autocomplete="off"
                        name="unit"
                      >
                        <option
                          value=""
                        />
                        <option
                          value="cup"
                        >
                          cup
                        </option>
                        <option
                          value="package"
                        >
                          package
                        </option>
                        <option
                          value="ounce"
                        >
                          ounce
                        </option>
                      </select>
                    </div>
                  </td>
                  <td>
                    <div>
                      <input
                        autocomplete="off"
                        name="ingredient"
                        placeholder="Ingredient"
                        required=""
                        type="text"
                        value="shredded sharp cheddar cheese"
                      />
                    </div>
                  </td>
                  <td>
                    <button
                      class="Button-choice-delete"
                      type="button"
                    >
                      Delete
                    </button>
                  </td>
                </tr>
                <tr
                  class="table-contentRow"
                >
                  <td
                    class="Table-amount"
                  >
                    <div>
                      <input
                        autocomplete="off"
                        min="0"
                        name="amount"
                        step="any"
                        type="number"
                        value="0.5"
                      />
                    </div>
                  </td>
                  <td>
                    <div>
                      <select
                        autocomplete="off"
                        name="unit"
                      >
                        <option
                          value=""
                        />
                        <option
                          value="cup"
                        >
                          cup
                        </option>
                        <option
                          value="package"
                        >
                          package
                        </option>
                        <option
                          value="ounce"
                        >
                          ounce
                        </option>
                      </select>
                    </div>
                  </td>
                  <td>
                    <div>
                      <input
                        autocomplete="off"
                        name="ingredient"
                        placeholder="Ingredient"
                        required=""
                        type="text"
                        value="salt"
                      />
                    </div>
                  </td>
                  <td>
                    <button
                      class="Button-choice-delete"
                      type="button"
                    >
                      Delete
                    </button>
                  </td>
                </tr>
                <tr
                  class="table-contentRow"
                >
                  <td
                    class="Table-amount"
                  >
                    <div>
                      <input
                        autocomplete="off"
                        min="0"
                        name="amount"
                        step="any"
                        type="number"
                        value="0.5"
                      />
                    </div>
                  </td>
                  <td>
                    <div>
                      <select
                        autocomplete="off"
                        name="unit"
                      >
                        <option
                          value=""
                        />
                        <option
                          value="cup"
                        >
                          cup
                        </option>
                        <option
                          value="package"
                        >
                          package
                        </option>
                        <option
                          value="ounce"
                        >
                          ounce
                        </option>
                      </select>
                    </div>
                  </td>
                  <td>
                    <div>
                      <input
                        autocomplete="off"
                        name="ingredient"
                        placeholder="Ingredient"
                        required=""
                        type="text"
                        value="coarsely ground pepper"
                      />
                    </div>
                  </td>
                  <td>
                    <button
                      class="Button-choice-delete"
                      type="button"
                    >
                      Delete
                    </button>
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
        <div
          class="btn-group"
        >
          <button
            class="Button-navigation"
            style="opacity: 1;"
            type="submit"
          >
            Save
          </button>
          <button
            class="Button-navigation"
            type="button"
          >
            Cancel
          </button>
          <button
            class="Button-navigation"
          >
            Delete
          </button>
        </div>
      </form>
    </div>
  </div>
</div>
`;
